src/OpenLoco/Industry.cpp

```diff
@@ -213,4 +230,44 @@ namespace OpenLoco
         regs.dh = id();
         call(0x00454A43, regs);
     }
+
+    // 0x00459D43
+    void Industry::createMapAnimations()
+    {
+        for (size_t i = 0; i < numTiles; i++)
+        {
+            auto& tilePos = tiles[i];
+            auto baseZ = (tilePos.z & ~Location::null) / 4;
+            auto tile = TileManager::get(tilePos);
+
+            for (auto& el : tile)
+            {
+                auto industryEl = el.asIndustry();
+                if (industryEl == nullptr)
+                    continue;
+
+                if (industryEl->baseZ() != baseZ)
+                    continue;
+
+                auto tileIndustry = industryEl->industry();
+                if (tileIndustry != nullptr)
+                {
+                    auto industryObject = tileIndustry->object();
+                    if (industryObject != nullptr)
+                    {
+                        auto animOffsets = word_4F9274;
+                        if (industryObject->var_C6 & (1 << industryEl->var_6_1F()))
+                        {
+                            animOffsets = word_4F927C;
+                        }
+                        while (animOffsets[0].pos.x != Location::null)
+                        {
+                            TileManager::createAnimation(3, animOffsets->pos + tilePos, baseZ);
+                            animOffsets++;
+                        }
+                    }
+                }
```

Comment:
I would prefer we invert the logic to avoid nesting.
```suggestion
                if (tileIndustry == nullptr)
                    continue;

                auto industryObject = tileIndustry->object();
                if (industryObject == nullptr)
                    continue;

                auto animOffsets = word_4F9274;
                if (industryObject->var_C6 & (1 << industryEl->var_6_1F()))
                {
                    animOffsets = word_4F927C;
                }
                while (animOffsets[0].pos.x != Location::null)
                {
                    TileManager::createAnimation(3, animOffsets->pos + tilePos, baseZ);
                    animOffsets++;
                }
```

Reply:
This is because my coding style is single entry, single exit for the most part. https://www.tomdalling.com/blog/coding-tips/coding-tip-have-a-single-exit-point/

It is just a personal preference.

Reply:
This comes from a age where explicit resource management was required in languages such as C and having a single exit made it easier to manage this, this was relevant in the 70's maybe.

It just makes it a lot harder to read and the optimization/resource concerns dont exist these days.